### PR TITLE
Feature 182 bucket tags - Tile UI editor

### PR DESCRIPTION
**Pull request by kirillston**
Catalog service description presented in tile UI has new feature - bucket tags editor for Service and Plan definitions (https://github.com/thecodeteam/ecs-broker-tile/pull/33).

This PR includes changes of initial Service Properties processing

### CODE REVIEW

File: src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java [hunk 1]
```diff
@@ -160,6 +162,37 @@ public void setServiceSettings4(String serviceSettingsJson) throws IOException {
...
+        if (bucketTagsString == null) {
+            return settings;
+        }
+
+        String[] tagsPairs = bucketTagsString.split(BUCKET_TAG_PAIRS_DELIMITER);
+
+        for(String tagPair: tagsPairs) {
+            String[] tag = tagPair.split(BUCKET_TAG_PAIR_KEY_VALUE_DELIMITER);
+            if (tag.length != 2) {
+                throw new ServiceBrokerException("Invalid bucket tag passed. Enable to split '" + tagPair +
```
Comment (itisevgeny) [inline src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java:179]: Unable?

File: src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java [hunk 2]
```diff
@@ -160,6 +162,37 @@ public void setServiceSettings4(String serviceSettingsJson) throws IOException {
...
+    private Map<String, Object> parseBucketTags(Map<String, Object> settings) {
+        List<Map<String, String>> bucketTags = new ArrayList<>();
+
+        String bucketTagsString = (String) settings.get(BUCKET_TAGS);
+
+        if (bucketTagsString == null) {
+            return settings;
+        }
+
+        String[] tagsPairs = bucketTagsString.split(BUCKET_TAG_PAIRS_DELIMITER);
```
Comment (rogalv) [inline src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java:174]: You have regexp checks in tile inputs, which can be reused here to validate input before splitting

File: src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java [hunk 3]
```diff
@@ -160,6 +162,37 @@ public void setServiceSettings4(String serviceSettingsJson) throws IOException {
         return settings;
     }
 
+    private Map<String, Object> parseBucketTags(Map<String, Object> settings) {
```
Comment (rogalv) [inline src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java:165]: Do we have test for tags parse?

File: src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java (hunk 1 above)
Comment (kirillston) [reply src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java]: Sure :) My typo

File: src/main/java/com/emc/ecs/servicebroker/service/EcsService.java [hunk 4]
```diff
@@ -702,6 +703,48 @@ public DataServiceReplicationGroup lookupReplicationGroup(String replicationGrou
                 .orElseThrow(() -> new ServiceBrokerException("ECS replication group not found: " + replicationGroup));
     }
 
+    static List<Map<String, String>> mergeBucketTags(ServiceDefinitionProxy service, PlanProxy plan, Map<String, Object> requestParameters) {
```
[the remark this answers is not in the document]
Comment (kirillston) [reply src/main/java/com/emc/ecs/servicebroker/service/EcsService.java:713]: Done. Added short description on mergeBucketTags() method

File: src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kirillston) [reply src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java]: My typo. Fixed

File: src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java (hunk 2 above)
Comment (kirillston) [reply src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java:188]: Added several regex-based checks on correct bucket tags format

File: src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java (hunk 3 above)
Comment (kirillston) [reply src/main/java/com/emc/ecs/servicebroker/config/CatalogConfig.java]: Added test-case on different input settings of method

**Comment by kirillston** [comment]
@itisevgeny , unfortunately, I cannot re-request your review supposing in cause of your membership status

**Pull request merged**